packages/block-library/src/image/edit.js

```diff
@@ -110,10 +110,15 @@ export function ImageEdit( {
 		metadata,
 	} = attributes;
 	const [ temporaryURL, setTemporaryURL ] = useState( attributes.blob );
-	const figureRef = useRef();
 
-	const [ contentResizeListener, { width: containerWidth } ] =
-		useResizeObserver();
+	const containerRef = useRef();
+	// Only observe the max width from the parent container when the parent layout is default or constrained.
+	// This won't work for flex or grid layouts because the container width changes with image.
+	// TODO: Find a way to observe the container width for flex and grid layouts.
+	const isMaxWidthContainerWidth =
+		parentLayout?.type === 'default' ||
+		parentLayout?.type === 'constrained';
```

Comment:
I think this might need to also include an absence of a parent layout type. Or to put it another way, what if we checked that the parent layout exists (is truthy) and isn't `flex` or `grid`?

I ran into this issue because in a post with a Columns block where a user hasn't yet interacted with the layout settings for an individual Column, the `parentLayout` will be an object but won't have an explicit `type` property set, as it's _implicitly_ a flow / default layout rather than that being explicit:

<img width="1322" alt="image" src="https://github.com/user-attachments/assets/c4e520b5-9252-4401-ad3e-f21f9eb63c19">

In that case the parentLayout object was: `{contentSize: '620px', wideSize: '1280px'}`

The problem I ran into is that with the current check, an Image in the Column block no longer gets the max-width behaviour:

https://github.com/user-attachments/assets/3c170d9b-7d02-46c2-918a-2e4ee2cfeec0

Note: if a user has toggled the layout on the individual Column block, then they _might_ get the max-width behaviour, because the `type` will be set explicitly: `{contentSize: '620px', wideSize: '1280px', type: 'default'}`. So this issue is to do with Column blocks where the layout hasn't been touched since the block was added to the post.

Reply:
Good catch! I think this makes sense! I didn't know about this detail about layouts 😅 (a proof that TypeScript could be useful here?). Committed in [5e1e199](https://github.com/WordPress/gutenberg/pull/64819/commits/5e1e19909511f21276aee37c684a3d1f174eca24).